general-superstaq/general_superstaq/superstaq_client.py

```diff
@@ -797,15 +824,17 @@ def _make_request(self, request: Callable[[], requests.Response]) -> requests.Re
             TimeoutError: If the requests retried for more than `max_retry_seconds`.
 
         Returns:
-            The `request.Response` from the final successful request call.
+            The response from the final successful request call.
         """
         # Initial backoff of 100ms.
         delay_seconds = 0.1
         while True:
             try:
                 response = request()
+
+                response_content = self._handle_response(response)
                 if response.ok:
-                    return response
+                    return response_content
```

Comment:
thanks for noticing/fixing this! one thought: we probably only want to print warnings in the event of non-retriable status codes. what if we just replaced these lines in `_handle_status_codes`: https://github.com/Infleqtion/client-superstaq/blob/9c99804ba2947ca9139f79ef1347949d39660243/general-superstaq/general_superstaq/superstaq_client.py#L756-L759
to
```python
        try:
            json_content = self._handle_response(response)
        except requests.JSONDecodeError:
            json_content = None
```
(and then reverted the change to `_handle_response`, which shouldn't be necessary anymore)